src/monster.cpp

```diff
@@ -1024,11 +1016,41 @@ void Monster::onThinkYell(uint32_t interval)
 	}
 }
 
+bool Monster::walkToSpawn()
+{
+	if (walkingToSpawn || !spawn || !targetList.empty()) {
+		return false;
+	}
+
+	int32_t distance = std::max(Position::getDistanceX(position, masterPos), Position::getDistanceY(position, masterPos));
+	if (distance == 0) {
+		return false;
+	}
+
+	std::vector<Direction> dirList;
+	if (!getPathTo(masterPos, dirList, 0, std::max<int32_t>(0, distance - 5), true, true, distance)) {
```

Comment:
Last argument, `distance`, should probably be removed, forgot to cleanup. Will do tomorrow. 💩

Reply:
It actually has to stay.